### PR TITLE
Add yaml for UserSignup CRD and example

### PR DESCRIPTION
**Pull request by sbryzak**
Upstream issue: https://jira.coreos.com/browse/CRT-225

This PR updates the CRD and example yaml to add the username field to the `UserSignup` spec.

Related PR https://github.com/codeready-toolchain/api/pull/31

Signed-off-by: Shane Bryzak <sbryzak@gmail.com>

### CODE REVIEW

File: deploy/crds/toolchain_v1alpha1_usersignup.yaml [hunk 1]
```diff
@@ -11,6 +11,10 @@ spec:
     name: User ID
     priority: 1
     type: string
+  - JSONPath: .spec.username
+    name: Username
+    priority: 1
+    type: string
```
Comment (MatousJobanek) [inline deploy/crds/toolchain_v1alpha1_usersignup.yaml]: this should be generated from `api` repo - see my comment there: https://github.com/codeready-toolchain/api/pull/31#pullrequestreview-282615206

Comment (MatousJobanek) [inline deploy/crds/toolchain_v1alpha1_usersignup.yaml]: this should be generated from `api` repo - see my comment there: https://github.com/codeready-toolchain/api/pull/31#pullrequestreview-282615206

File: deploy/crds/toolchain_v1alpha1_usersignup.yaml [hunk 2]
```diff
@@ -66,8 +70,13 @@ spec:
               description: UserID is the user ID from RHD Identity Provider token
                 (“sub” claim)
               type: string
+            username:
+              description: Username is the original, unaltered username for the
+                user's RHD account.
```
Comment (MatousJobanek) [inline deploy/crds/toolchain_v1alpha1_usersignup.yaml]: this should contain the generated description from `api` repo.

File: deploy/crds/toolchain_v1alpha1_usersignup.yaml (hunk 1 above)
Comment (sbryzak) [reply deploy/crds/toolchain_v1alpha1_usersignup.yaml]: When I run `make generate` in the api project, I get the following error:

`The local '../host-operator' repository has pending changes. Please stash them or commit them, first.
`

There are no uncommitted changes, any idea what this means?

Comment (sbryzak) [reply deploy/crds/toolchain_v1alpha1_usersignup.yaml]: Nevermind, I worked it out.. simply having a file in the host-operator directory (I usually pipe my test output to a text file) was enough to break this make target, even though the file wasn't under git version control.  I've regenerated the crd and added the changes to the PR.

Comment (MatousJobanek) [reply deploy/crds/toolchain_v1alpha1_usersignup.yaml]: well, the make target checks the git status using `git status -s ` so if the file was detected, then the target failed as it expected a clear environment

**Comment by MatousJobanek** [comment]
@sbryzak don't forget to change the e2e tests: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/codeready-toolchain_host-operator/61/pull-ci-codeready-toolchain-host-operator-master-e2e/226

File: go.mod [hunk 3]
```diff
@@ -34,6 +28,8 @@ require (
 )
 
 replace (
+	git.apache.org/thrift.git => github.com/apache/thrift v0.12.0
```
Comment (MatousJobanek) [inline go.mod]: this is already part of the master https://github.com/codeready-toolchain/host-operator/blob/master/go.mod#L37

File: go.mod [hunk 4]
```diff
@@ -1,24 +1,18 @@
...
 
 require (
-	cloud.google.com/go v0.40.0 // indirect
-	github.com/Azure/go-autorest v13.0.0+incompatible // indirect
 	github.com/appscode/jsonpatch v0.0.0-20190625103638-320dcdd0e1f7 // indirect
-	github.com/codeready-toolchain/api v0.0.0-20190813211537-c2a9a838ce38
-	github.com/codeready-toolchain/toolchain-common v0.0.0-20190830131634-231d1bd08d5f
+	github.com/codeready-toolchain/api v0.0.0-20190906003527-443e8ceac134
+	github.com/codeready-toolchain/toolchain-common v0.0.0-20190904115454-79e74322be0a
+	github.com/codeready-toolchain/toolchain-e2e v0.0.0-20190904123258-c90f69f65a94 // indirect
```
Comment (MatousJobanek) [inline go.mod]: This shouldn't be here at all

File: go.mod (hunk 3 above)
Comment (sbryzak) [reply go.mod]: I've merged master

File: go.mod (hunk 4 above)
Comment (sbryzak) [reply go.mod]: This was cleaned up once I ran go mod tidy.

**Pull request merged**